tests/suite/cli_misc.rs

```diff
@@ -667,27 +667,19 @@ fn rename_rls_remove() {
 fn toolchain_broken_symlink() {
     use std::fs;
     use std::path::Path;
-
-    #[cfg(unix)]
-    fn create_symlink_dir<P: AsRef<Path>, Q: AsRef<Path>>(src: P, dst: Q) {
-        use std::os::unix::fs;
-        fs::symlink(src, dst).unwrap();
-    }
-
-    #[cfg(windows)]
-    fn create_symlink_dir<P: AsRef<Path>, Q: AsRef<Path>>(src: P, dst: Q) {
-        use std::os::windows::fs;
-        fs::symlink_dir(src, dst).unwrap();
-    }
+    use rustup::utils::raw::symlink_dir;
 
     clitools::test(Scenario::None, &|config| {
         // We artificially create a broken symlink toolchain -- but this can also happen "legitimately"
         // by having a proper toolchain there, using "toolchain link", and later removing the directory.
         fs::create_dir(config.rustupdir.join("toolchains")).unwrap();
-        create_symlink_dir(
-            config.rustupdir.join("this-directory-does-not-exist"),
-            config.rustupdir.join("toolchains").join("test"),
-        );
+        fs::create_dir(config.rustupdir.join("this-directory-does-not-exist")).unwrap();
+        symlink_dir(
+            &config.rustupdir.join("this-directory-does-not-exist"),
+            &config.rustupdir.join("toolchains").join("test"),
+        ).unwrap();
+        fs::remove_dir(config.rustupdir.join("this-directory-does-not-exist")).unwrap();
```

Comment:
`symlink_dir()` calls `fs::canonicalize()`, which requires the path to actually exist, so I had to create the directory and delete it afterwards.